src/thrift_parser.yrl

```diff
@@ -0,0 +1,238 @@
+definition -> enum_def: '$1'.
+definition -> exception_def: '$1'.
+definition -> typedef_def: '$1'.
+definition -> struct_def: '$1'.
+definition -> union_def: '$1'.
+definition -> service_def: '$1'.
+
+set_type -> set lt type gt: '$3'.
+map_type -> map lt type comma type gt: {'$3', '$5'}.
+list_type -> list lt type gt: '$3'.
+
+type -> base_type: '$1'.
+type -> set_type: {set, '$1'}.
+type -> map_type: {map, '$1'}.
+type -> list_type: {list, '$1'}.
+type -> ident: 'Elixir.Thrift.Parser.Models.StructRef':new(unwrap('$1')).
+
+base_type -> bool: bool.
+base_type -> byte: i8.
+base_type -> i8: i8.
+base_type -> i16: i16.
+base_type -> i32: i32.
+base_type -> i64: i64.
+base_type -> double: double.
+base_type -> string: string.
+base_type -> binary: binary.
+
+include_def ->
+    include ident:
+        'Elixir.Thrift.Parser.Models.Include':new(unwrap('$2')).
+
+namespace_def ->
+    namespace ns_name ident:
+        'Elixir.Thrift.Parser.Models.Namespace':new('$2', unwrap('$3')).
+
+constant_def ->
+    const type ident equals literal:
+        'Elixir.Thrift.Parser.Models.Constant':new(unwrap('$3'), '$5', '$2').
+
+ns_name -> asterisk: "*".
+ns_name -> ident: unwrap('$1').
+
+
+%% JS Style mapping "foo": 32
+mapping -> literal colon literal: {'$1', '$3'}.
+mappings -> mapping: ['$1'].
+mappings -> mapping comma mappings: ['$1'] ++ '$3'.
+
+% A list of literals ["hi", "bye", 3, 4]
+literal_list -> literal: ['$1'].
+literal_list -> literal comma literal_list: ['$1'] ++ '$3'.
+
+literal -> int: unwrap('$1').
+literal -> double: unwrap('$1').
+literal -> string: unwrap('$1').
+literal -> open_curly literal_list close_curly: '$2'.
+literal -> open_curly mappings close_curly: '$2'.
+literal -> open_bracket literal_list close_bracket: '$2'.
+
+field_sep -> '$empty': nil.
+field_sep -> comma: nil.
+field_sep -> semi: nil.
+
+enum_def ->
+    enum ident open_curly enum_body close_curly:
+        'Elixir.Thrift.Parser.Models.TEnum':new(unwrap('$2'), '$4').
+enum_body -> enum_value field_sep: ['$1'].
+enum_body -> enum_value field_sep enum_body: ['$1'] ++ '$3'.
+
+enum_value -> ident: unwrap('$1').
+enum_value -> ident equals int: {unwrap('$1'), unwrap('$3')}.
+
+exception_def ->
+    exception ident open_curly fields close_curly:
+        'Elixir.Thrift.Parser.Models.Exception':new(unwrap('$2'), '$4').
+
+fields -> '$empty': [].
+fields -> field_spec field_sep fields: ['$1'] ++ '$3'.
+
+field_id -> int colon: unwrap('$1').
+field_id -> '$empty': nil.
+
+field_spec ->
+    field_id field_required type ident field_default:
+        'Elixir.Thrift.Parser.Models.Field':new('$1', '$2', '$3', unwrap('$4'), '$5').
+
+field_required -> required: true.
+field_required -> optional: false.
+field_required -> '$empty': default.
+
+field_default -> '$empty': nil.
+field_default -> equals literal: '$2'.
+
+typedef_def ->
+    typedef type ident:
+        {typedef, '$2', unwrap('$3')}.
+
+struct_def ->
+    struct ident open_curly fields close_curly:
```

Comment:
This might read better with the literal `'{`'` and `'}'` characters. What are the tradeoffs between that and using tokens from the lexer?